pkg/proto/configuration/global/global.proto

```diff
@@ -56,18 +57,96 @@ message PrometheusPushgatewayConfiguration {
   google.protobuf.Duration push_interval = 5;
 }
 
+message OpenCensusConfiguration {
+  // Export data to an OpenCensus agent. The recommended configuration
+  // is to use opentelemetry-collector and enable its opencensus-agent
+  // receiver.
+
+  // Connection address for the agent. Default is localhost:55678.
+  string address = 1;
+
+  // TLS configuration. TLS is not enabled when left unset.
+  buildbarn.configuration.tls.ClientConfiguration tls = 2;
+
+  // OpenCensus service name.
+  string service_name = 3;
+}
+
+message OpenTelemetryConfiguration {
+  // Connection address for the collector. Default is localhost:55680.
+  string address = 1;
+
+  // TLS configuration. TLS is not enabled when left unset.
+  buildbarn.configuration.tls.ClientConfiguration tls = 2;
+
+  // Resource detectors. Labels will be merged, with earlier things in
+  // this entries being preferred over later ones.
+  enum ResourceDetector {
+    FROM_ENVIRONMENT_VARIABLE =
+        0;    // Use OTEL_RESOURCE_ATTRIBUTES environment variable
+    AWS = 1;  // Use AWS instance metadata
+    GCE = 2;  // Use GCE instance metadata
+    GKE = 3;  // Use GKE API
+  }
+  repeated ResourceDetector resource_detectors = 3;
+
+  // Queue size for buffering spans. Default is 2048.
+  int32 max_queue_size = 4;
+
+  // If true, then span Start() and End() functions will block when
+  // the queue is full. This will significantly degrade performance,
+  // but does guarantee no spans are lost, which can be useful when
+  // debugging with the sampling probability set to 1.0.
+  bool block_on_queue_full = 5;
+
+  // Maximum duration to hold a batch before sending. Default is 5s.
+  google.protobuf.Duration batch_timeout = 6;
+
+  // Maximum number of spans to batch before sending. Default is 512.
+  int32 max_export_batch_size = 7;
+}
+
 message TracingConfiguration {
-  // Jaeger configuration for tracing.
-  JaegerConfiguration jaeger = 1;
+  // Jaeger configuration for tracing has been replaced by oc and otel.
+  reserved 1;
 
-  // Stackdriver Trace configuration for tracing.
-  StackdriverConfiguration stackdriver = 2;
+  // Stackdriver Trace configuration has been replaced by oc and otel.
+  reserved 2;
 
   // Export stats on traces to Prometheus.
   bool enable_prometheus = 3;
```

Comment:
Is my observation correct that this option only applies to just OpenCensus and not OpenTracing? If so, would it make sense to move it into the relevant nested message?

Reply:
Hmm, interesting question. It is currently true that it's only wired up to OpenCensus. That's mostly because the OpenTelemetry version of this isn't very complete yet (tracing is much further ahead) so I didn't have a sensible migration path right now. At some point in the future, it should be handled on the otel side... does it make more sense to leave it here as a generic config knob that's somehow implemented, or to move it around as the code does?

Reply:
Sorry for not responding to this earlier.

I'd suggest we move it around as the code does, right? That both keeps the code self-contained. One added 'advantage' is that it causes startup to fail if people refer to OpenCensus by the time we remove support for it entirely. It's a good way to signal people of changes.

Reply:
Done